File: client/transcenders_protocol.js
export const MAX_NAME_LENGTH = 8
export const MAX_TRANSFER_BUFFER_LENGTH=32 

export const Protocol = {
    Reserved: {
	pingPong: 0x1,
	joinNew: 0x2,
	joinExisting: 0x3,
	error: 0x4,
	deleteRequest: 0x5
    },
    Usable: {
	player: 0x9,
	enemyThreat: 0xA,
	weapon: 0xB,
	item: 0xC
    }
}

File: server/transcenders_protocol.cpp
struct protocol {
    /**
      * [0x1, 0x8]
      */
    struct reserved {
	static const uint8_t ping_pong = 0x1;
	static const uint8_t join_new = 0x2;
	static const uint8_t join_existing = 0x3;
	static const uint8_t error = 0x4;
	static const uint8_t delete_request = 0x5;
    };

    /**
      * [0x9, 0x20]
      */
    struct usable {
	static const uint8_t player = 0x9;
	static const uint8_t enemy_threat = 0xA;
	static const uint8_t weapon = 0xB;
	static const uint8_t item = 0xC;
    };
};
